src/writer/vault.py

```
@@ -0,0 +1,45 @@
+        self.secrets: Union[Dict, None] = None
+
+    def get_secrets(self) -> Dict:
+        if self.secrets is None:
+            self.secrets = self._fetch()
+        return self.secrets
+
+    def refresh(self):
+        self.secrets = self._fetch()
+
+    def _fetch(self) -> Dict:
+        # TODO: move the API call to a service
```

Pylint (3.3.7)

[warning] TODO: move the API call to a service

(W0511)